src/ppom-controller.ts

```diff
@@ -345,129 +361,216 @@ export class PPOMController extends BaseControllerV2<
+   * Returns an array of new files that should be downloaded and saved to storage.
+   * The function is invoked if user if attempting transaction for a network,
+   * for which data is not previously fetched.
    *
-   * If the ppom configuration is out of date, this function will call `updatePPOM`
-   * to update the configuration.
+   * @returns A promise that resolves to an array of new files to download and save to storage.
    */
-  async #maybeUpdatePPOM() {
-    if (await this.#shouldUpdate()) {
-      await this.#updatePPOM();
+  async #getNewFilesForCurrentChain(): Promise<void> {
+    const { chainId, versionInfo } = this.state;
+    for (const fileVersionInfo of versionInfo) {
+      //  download all files for the current chain.
+      if (fileVersionInfo.chainId !== chainId) {
+        continue;
+      }
+
+      await this.#getFile(fileVersionInfo);
     }
+    this.#setChainIdDataFetched(chainId);
   }
 
-  /*
-   * Fetch the blob from the PPOM cdn.
+  /**
+   * Returns an array of new files that should be downloaded and saved to storage.
+   *
+   * @returns A promise that resolves to an array of new files to download and save to storage.
    */
-  async #fetchBlob(fileUrl: string): Promise<ArrayBuffer> {
-    const response = await safelyExecute(
-      async () => fetch(fileUrl, { cache: 'no-cache' }),
-      true,
+  async #getNewFilesForAllChains(): Promise<void> {
+    const {
+      chainIdCache,
+      storageMetadata,
+      versionInfo: stateVersionInfo,
+    } = this.state;
+
+    // create a map of chainId and files belonging to that chainId
+    const chainIdsFileInfoList = chainIdCache.map(
+      (chain): { chainId: string; versionInfo: PPOMFileVersion[] } => ({
+        chainId: chain.chainId,
+        versionInfo: stateVersionInfo.filter(
+          (versionInfo) =>
+            versionInfo.chainId === chain.chainId &&
+            !this.#checkFilePresentInStorage(storageMetadata, versionInfo),
+        ),
+      }),
     );
 
-    switch (response?.status) {
-      case 200: {
-        return await response.arrayBuffer();
-      }
+    // clear already scheduled fetch if any
+    if (this.#fileScheduleInterval) {
+      clearInterval(this.#fileScheduleInterval);
+    }
 
-      default: {
-        throw new Error(`Failed to fetch file with url ${fileUrl}`);
+    // build a list of files to be fetched for all networks
+    const fileToBeFetchedList: {
+      fileVersionInfo: PPOMFileVersion;
+      isLastFileOfNetwork: boolean;
+    }[] = [];
+    chainIdsFileInfoList.forEach((chainIdFileInfo) => {
+      const { chainId, versionInfo } = chainIdFileInfo;
+      versionInfo.forEach((fileVersionInfo, index) => {
+        fileToBeFetchedList.push({
+          fileVersionInfo,
+          isLastFileOfNetwork: index === versionInfo.length - 1,
+        });
+      });
+      if (versionInfo.length === 0) {
+        // set dataFetched to true for chainId
+        this.#setChainIdDataFetched(chainId);
       }
+    });
```

Comment:
Would it also help break up this method to move all the logic up this point into a `getNewFilesForAllChains` where `get` literally means return as we'd use it to just identify the array of file info?